iota/commands/extended/get_new_addresses.py

```diff
@@ -36,25 +36,35 @@ def _execute(self, request):
     count           = request['count'] # type: Optional[int]
     index           = request['index'] # type: int
     security_level  = request['securityLevel'] # type: int
+    checksum        = request['checksum'] # type: Optional[bool]
     seed            = request['seed'] # type: Seed
 
     return {
-      'addresses': self._find_addresses(seed, index, count, security_level),
+      'addresses': 
+         self._find_addresses(seed, index, count, security_level, checksum),
     }
 
-  def _find_addresses(self, seed, index, count, security_level):
-    # type: (Seed, int, Optional[int], int) -> List[Address]
+  def _find_addresses(self, seed, index, count, security_level, checksum):
+    # type: (Seed, int, Optional[int], int, Optional[bool]) -> List[Address]
     """
     Find addresses matching the command parameters.
     """
-    # type: (Seed, int, Optional[int]) -> List[Address]
-    generator = AddressGenerator(seed, security_level)
+    # type: (Seed, int, Optional[bool]) -> List[Address]
+    generator = AddressGenerator(seed, security_level, checksum)
 
     if count is None:
       # Connect to Tangle and find the first address without any
       # transactions.
       for addy in generator.create_iterator(start=index):
-        response = FindTransactionsCommand(self.adapter)(addresses=[addy])
+        # If we're generating addresses with checksums we need to check
+        # for transactions on the address without the checksum
+        if not checksum:
+          response = FindTransactionsCommand(self.adapter)(addresses=[addy])
+        else:
+          response = FindTransactionsCommand(self.adapter)(
+            addresses=[addy][:-AddressChecksum.LEN]
```

Comment:
Alternatively, we could use `addy.address` here.

https://github.com/iotaledger/iota.lib.py/blob/2.0.2/iota/types.py#L698-L699

Reply:
Aha, I was looking for that! I don't know why I didn't even guess it. Much cleaner.